openedx/adg/lms/applications/serializers.py

```diff
@@ -0,0 +1,105 @@
+"""
+Serializers for applications app
+"""
+from rest_framework import serializers
+
+from .helpers import check_validations_for_current_record, check_validations_for_past_record
+from .models import Education, WorkExperience
+
+
+class EducationSerializer(serializers.ModelSerializer):
+    """
+    Serializer for the `Education` model
+    """
+
+    def validate(self, attrs):
+        """
+        Custom validations on education attrs
+
+        Arguments:
+            attrs (dict): Dictionary containing education attributes
+
+        Returns:
+            dict: Returns updated education attributes after validation or raises validation exceptions
+        """
+        attrs, errors = self._check_area_of_study_validations(attrs)
+
+        if attrs.get('is_in_progress'):
+            errors.update(
+                check_validations_for_current_record(
+                    attrs,
+                    '{key} isn\'t applicable for degree in progress', 'date_completed_month', 'date_completed_year'
+                )
+            )
+        else:
+            errors.update(
+                check_validations_for_past_record(
+                    attrs, '{key} is required for past degree', 'date_completed_month', 'date_completed_year'
+                )
+            )
+        if errors:
+            raise serializers.ValidationError(errors)
+
+        return attrs
+
+    def _check_area_of_study_validations(self, attrs):
+        """
+        Area of study validations, check if `area_of_study` is provided for all degrees except for `HD` if it isn't
+        raises exception. For `HD` if value of `area_of_study` is provided replace it with empty string`''`.
```

Comment:
1. `Validate area of study by checking if...`

2. Lets have a complete name for `HD` so that it is more readable 

3. `...if it isn't raising exception.` or `...if it doesn't raise exception.`